server/rest.go

```diff
@@ -788,6 +788,35 @@ func (s *Server) AddressHolders(c echo.Context) error {
 		krcTokenInfo, _ := s.getKRCTokenInfo(ctx, holders[i].ContractAddress)
 		if krcTokenInfo != nil {
 			holders[i].Logo = krcTokenInfo.Logo
+			// get holder balance from RPC for rechecking
+			if strings.EqualFold(krcTokenInfo.TokenType, cfg.SMCTypeKRC20) {
+				smcABIStr, err := s.cacheClient.SMCAbi(ctx, cfg.SMCTypePrefix+cfg.SMCTypeKRC20)
+				if err != nil {
+					s.logger.Warn("Cannot get KRC20 token ABI", zap.Error(err), zap.String("smcAddress", holders[i].ContractAddress))
+					continue
+				}
+				smcABI, err := s.decodeSMCABIFromBase64(ctx, smcABIStr, holders[i].ContractAddress)
+				if err != nil {
+					s.logger.Warn("Cannot decode KRC20 token ABI", zap.Error(err), zap.String("smcAddress", holders[i].ContractAddress))
+					continue
+				}
```

Comment:
Why we need those lines here? Its KRC20 for sure after this check
```
strings.EqualFold(krcTokenInfo.TokenType, cfg.SMCTypeKRC20)
```

Reply:
I will change this check if token type is start with `"KRC"`, because many KRC types have holders and their balances.

Reply:
Okay I removed the check, its redundant